plugins/modules/server.py

```diff
@@ -564,10 +564,20 @@ def _update_server(self) -> None:
         try:
             previous_server_status = self.hcloud_server.status
 
+            update_params = {}
+
+            name = self.module.params.get("name")
+            if name is not None and self.hcloud_server.name != name:
+                self.module.fail_on_missing_params(required_params=["id"])
```

Comment:
Should this requirement be added to the documentation?

Reply:
I think the pattern to renamed a resource is well understood and the error is be clear enough to educate the users.